aspnetcore/fundamentals/http-context/samples/6.x/HttpContextInBackgroundThread/EmailService.cs

```diff
@@ -1,31 +1,37 @@
 namespace HttpContextInBackgroundThread;
 
+public interface IEmailService
+{
+    void SendEmail(string email);
+}
+
 public class EmailService : IEmailService
 {
     private readonly IHttpContextAccessor _httpContextAccessor;
     private readonly ILogger<EmailService> _logger;
-    private const string? BackgroundCorrelationId = "background-correlation-id";
+    private const string UserAgent = "Unknown";
 
-    public EmailService(IHttpContextAccessor httpContextAccessor, ILogger<EmailService> logger)
+    public EmailService(IHttpContextAccessor httpContextAccessor,
+                                     ILogger<EmailService> logger)
     {
         _httpContextAccessor = httpContextAccessor;
         _logger = logger;
     }
 
     public void SendEmail(string email)
     {
-        // Services should account for the possibility of HttpContext being null if not called from a request thread.
-        // e.g. When a service is called from a BackgroundWorker the HttpContext is null.
-        // The service should either throw or gracefully handle this.
+        var request = _httpContextAccessor.HttpContext?.Request;
+        var userAgent = request?.Headers["user-agent"].ToString()
+                                         ?? UserAgent;
 
-        var correlationId = _httpContextAccessor.HttpContext?.Request.Headers["X-Correlation-Id"].ToString() ?? BackgroundCorrelationId;
-
-        _ = SendEmailCoreAsync(correlationId);
+        _ = SendEmailCoreAsync(userAgent);
```

Comment:
`_ = SomeAsyncMethod();` is almost always code smell which I feel bad showing in a best-practices doc.

It means the host won't wait on the background processing before disposing any service dependencies. This is a common source of ObjectDisposedExceptions during shutdown.

It also means any exceptions go unobserved. This is usually mitigated by wrapping the dispatched code in a giant try/catch and logging any Exceptions, but that doesn't help you if the logger provider's already been disposed because the service didn't track it's ongoing async calls and delay shutdown in DisposeAsync.

Unfortunately, doing background processing the right way is a lot more complicated. Really we should be doing something like what's shown in https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-6.0&tabs=visual-studio#queued-background-tasks. Can we just reference the BackgroundTaskQueue from there?

```suggestion
        await _taskQueue.QueueBackgroundWorkItemAsync(cancellationToken => SendEmailCoreAsync(userAgent, cancellationToken));
```